src/App.tsx

```diff
@@ -1,61 +1,110 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
 export const App = () => {
+  const users:User[] = usersFromServer;
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+  const [userId, setUserId] = useState<number>(0);
+  const [title, setTitle] = useState<string>('');
+  const [errorTitle, setErrorTitle] = useState<boolean>(false);
+  const [errorUser, setErrorUser] = useState<boolean>(false);
+
+  const addUser = () => {
+    const id = [...todos].sort(
+      (todo1, todo2) => todo2.id - todo1.id,
+    )[0].id + 1;
+
+    setTodos(
+      [
+        ...todos,
+        {
+          id,
+          title,
+          completed: false,
+          userId,
+        }],
+    );
+  };
+
+  const handleSubmitButton = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    if (!userId) {
+      setErrorUser(true);
+    }
+
+    if (!title.trim()) {
+      setErrorTitle(true);
+    }
+
+    if (userId && title) {
+      addUser();
+      setUserId(0);
+      setTitle('');
+    }
```

Comment:
Thanks, for all advices). But this one, I`ve particularly done this way, because in the situation when both title and username are undefined, it will setError only for one of the options.(

Reply:
hey! you can create separate validation function, check all errors there
and in your submit function 
```
if (!isValidData()) {
  return
}
```